demo/ios/OrcaDemo/OrcaDemo/ContentView.swift

```diff
@@ -0,0 +1,143 @@
+                    VStack {
+                        ScrollView {
+                            TextEditor(text: $text)
+                                .transparentScrolling()
+                                .padding()
+                                .frame(minWidth: 0,
+                                       maxWidth: .infinity,
+                                       minHeight: geometry.size.height,
+                                       maxHeight: .infinity)
+                                .font(.title3)
+                                .background(lightGray)
+                                .onChange(of: text) { _ in
+                                    text = String(text.prefix(Int(viewModel.maxCharacterLimit)))
+                                }
+                        }
+
+                        Text("\(text.count) / \(viewModel.maxCharacterLimit)")
+                            .font(.footnote)
+                            .frame(maxWidth: .infinity, alignment: .trailing)
+                            .foregroundColor(Color.gray)
+                    }
+                }
+
+                if viewModel.state == .INIT || viewModel.state == .READY {
+                    Text("Enter any text to be synthesized")
+                        .padding()
+                        .font(.body)
+                        .foregroundColor(Color.black)
+                } else if viewModel.state == .PROCESSING {
+                    Text("Processing text...")
+                        .padding()
+                        .font(.body)
+                        .foregroundColor(Color.black)
+                } else if viewModel.state == .PLAYING {
+                    Text("Playing audio")
+                        .padding()
+                        .font(.body)
+                        .foregroundColor(Color.black)
+                } else if viewModel.state == .SYNTHESIZE_ERROR {
+                    Text(viewModel.synthesizeError)
+                        .padding()
+                        .foregroundColor(Color.white)
+                        .frame(maxWidth: .infinity)
+                        .background(dangerRed)
+                        .font(.body)
+                        .opacity(viewModel.synthesizeError.isEmpty ? 0 : 1)
+                        .cornerRadius(10)
+                } else {
+                    Text(viewModel.errorMessage)
+                        .padding()
+                        .foregroundColor(Color.white)
+                        .frame(maxWidth: .infinity)
+                        .background(dangerRed)
+                        .font(.body)
+                        .opacity(viewModel.errorMessage.isEmpty ? 0 : 1)
+                        .cornerRadius(10)
+                }
+
+                Button(action: {
+                    viewModel.toggleSynthesize(text: text)
+                },
+                       label: {
+                    Text(viewModel.state == .PLAYING ? "Stop" : "Synthesize")
```

Comment:
this button jumps in size when it changes text - would looks a bit better if it was a set length